### ISSUE TITLE
Non-existant stack reference in stack_output resolver throws confusing error

### ISSUE THREAD

**Issue by siulcode** [cloudreach/sceptre#148]
When using the stack output resolver, I misspelt the reference stack name. e.g:
`!stack_output incorrect_stack_name::SomeOutput`
 This was the error that was thrown:

```
[2017-06-21 16:11:25] - sceptre.environment - Checking for circular dependencies...
Traceback (most recent call last):
  File "/usr/bin/sceptre", line 11, in <module>
    sys.exit(cli())
  File "/usr/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/sceptre/cli.py", line 93, in decorated
    return func(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/sceptre/cli.py", line 305, in launch_env
    response = env.launch()
  File "/usr/lib/python2.7/site-packages/sceptre/environment.py", line 124, in launch
    self._check_for_circular_dependencies(launch_dependencies)
  File "/usr/lib/python2.7/site-packages/sceptre/environment.py", line 335, in _check_for_circular_dependencies
    if stack_name in dependencies[dependency]:
KeyError: u'dev/mgo-etl-svc/iam_managed_policies'
```
It is confusing as there are a lot references to circular dependencies. A more appropriate error should be thrown such as "reference stack not found". To do this stack_output resolver may need to check if the config for the reference stack exists on initialisation.

**Comment by sathed** [comment]
You also get this same error (different stacktrace) when you specify a non existing config file.

Example:

    # Directory structure:
    aws-project/
    └── config
        └── dev
            ├── config.yaml
            └── ec2_mongo.yaml

    # Command:
    $ sceptre create-stack dev ec2-mongo

Throws:

```
Traceback (most recent call last):
  File "/usr/bin/sceptre", line 11, in <module>
    load_entry_point('sceptre==1.2.1', 'console_scripts', 'sceptre')()
  File "/usr/lib/python3.6/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.6/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.6/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.6/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/sceptre/cli.py", line 94, in decorated
    return func(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/sceptre/cli.py", line 246, in create_stack
    response = env.stacks[stack].create()
KeyError: 'ec2-mongo'
```

It would be nice to have a better explanation of what is going on.

**Comment by ngfgrant** [comment]
Resolved as part of #370
